### PR TITLE
feat: save published items

### LINKED ISSUE #1544: Allow users to save changes to published items
- Change `handleSavePublishedItemRequest` to dispatch a `saveItemRequest` instead of a  `deployItemRequest`

- If metadata changed do not send a transaction, save item in builder server (the metadata will be saved by the reviewer at the end using rescueItem)

- Change `handleSaveItemRequest` saga to stop checking if the item is published

- Unblock UI of published item (except for item rarity and collection name)

### PR DESCRIPTION
**Pull request by cazala**
Closes #1544

### CODE REVIEW

File: src/components/Modals/CreateItemModal/CreateItemModal.tsx [hunk 1]
```diff
@@ -746,43 +753,43 @@ export default class CreateItemModal extends React.PureComponent<Props, State> {
...
-                                onChange={this.handleItemChange}
-                                isDisabled={isAddingRepresentation}
-                              />
-                            </Section>
-                          ) : (
-                            this.renderFields()
-                          )}
-                        </>
-                      )}
+                          <>
```
Comment (fzavalia) [inline src/components/Modals/CreateItemModal/CreateItemModal.tsx:756]: I think this new whitespaces should not be here. (Locally, Prettier removes them for me)

File: src/modules/item/sagas.spec.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        .run({ silenceTimeout: true })
+    })
+
+    it('should throw if file size is larger than 2 MB', () => {
+      const item = {
+        name: 'valid name',
+        description: 'valid description'
+      } as Item
+      return expectSaga(itemSaga, builderAPI)
+        .provide([[matchers.call.fn(calculateFinalSize), Promise.resolve(MAX_FILE_SIZE + 1)]])
```
Comment (fzavalia) [inline src/modules/item/sagas.spec.ts:76]: You could also use `.provide([[call(calculateFinalSize, expectedItem, expectedContent), Promise.resolve(MAX_FILE_SIZE + 1)]])` To also validate that the function is receiving the expected parameters

File: src/modules/item/selectors.ts [hunk 3]
```diff
@@ -25,9 +25,9 @@ export const getWalletItems = createSelector<RootState, Item[], string | undefin
 )
 
 export const getAuthorizedItems = createSelector<RootState, Collection[], Item[], string | undefined, Item[]>(
-  getAuthorizedCollections,
-  getItems,
-  getAddress,
+  state => getAuthorizedCollections(state),
```
Comment (fzavalia) [inline src/modules/item/selectors.ts:28]: I'm curious about why it was changed to this

**Review by fzavalia — APPROVED** [review]
LGTM!
Comments are for some minor things.

File: src/modules/item/selectors.ts (hunk 3 above)
Comment (cazala) [reply src/modules/item/selectors.ts:28]: This was due to circular deps between the selectors. Passing `getAuthorizedCollections` as reference there would be undefined, but calling it within that lambda works fine on runtime.

**Review by fzavalia — APPROVED** [review]

**Pull request merged**